src/include/duckdb/common/types/row/tuple_data_layout.hpp

```diff
@@ -0,0 +1,113 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/common/types/row/tuple_data_layout.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/common/common.hpp"
+#include "duckdb/common/types/row/row_layout.hpp"
+#include "duckdb/common/types/validity_mask.hpp"
+#include "duckdb/execution/operator/aggregate/aggregate_object.hpp"
+#include "duckdb/planner/expression.hpp"
+
+namespace duckdb {
+
+class TupleDataLayout {
+public:
+	using Aggregates = vector<AggregateObject>;
+	using ValidityBytes = TemplatedValidityMask<uint8_t>;
+
+	//! Creates an empty TupleDataLayout
+	TupleDataLayout();
+	//! Create a copy of this TupleDataLayout
+	TupleDataLayout Copy() const;
+
+public:
+	//! Initializes the RowLayout with the specified types and aggregates to an empty RowLayout
```

Comment:
They're essentially the same, but `TupleDataLayout` inlines structs within the layout rather than storing a pointer and writing them to the heap.

This is temporary, of course. I want to phase out the old `Row*` stuff and replace it with `TupleData*` stuff, but this PR is big enough. Once everything is replaced, we can rename it back, but for now, both need to exist.